### DISCUSSION TITLE
Can `sqlalchemy.inspect(engine).get_columns(<table_name>)` return values already converted to Python types without previous definition of table in `sqlalchemy.Table`?

### DISCUSSION THREAD

**Discussion by HardMax71** [sqlalchemy/sqlalchemy#11689] [Usage Questions] [answered]
[sqlalchemy, sqlite]

Hi, I am working on a GUI app and need sqlalchemy reflection. Assume that we have `test.db` in same folder as `main.py`, and `test.db` contains following user-generated table:
```sql
-- auto-generated definition
create table authors
(
    author_id     INTEGER
        primary key autoincrement,
    name          TEXT not null,
    birth_date    DATE,
    nationality   TEXT    default 'Unknown',
    books_written INTEGER default 0,
    check (books_written >= 0)
);
```

Now for whatever reason I want to get data about columns from this table. So, I wrote following script:
```python
# main.py
from sqlalchemy import create_engine, inspect

engine = create_engine('sqlite:///test.db')

inspector = inspect(engine)
columns = inspector.get_columns('authors')

print(columns)
```

Result is:
```
[{'name': 'author_id', 'type': INTEGER(), 'nullable': True, 'default': None, 'primary_key': 1}, 
{'name': 'name', 'type': TEXT(), 'nullable': False, 'default': None, 'primary_key': 0},
 {'name': 'birth_date', 'type': DATE(), 'nullable': True, 'default': None, 'primary_key': 0},
 {'name': 'nationality', 'type': TEXT(), 'nullable': True, 'default': "'Unknown'", 'primary_key': 0},
 {'name': 'books_written', 'type': INTEGER(), 'nullable': True, 'default': '0', 'primary_key': 0}]
```

As you can see, `default` values for columns that are not strings (e.g. `books_written`) are strings, and default values that are strings, are double-quoted: `'default': "'Unknown'"`. 

My question is as follows: Has SQLAlchemy sort of flag or sth so that values in `default` will be returned already converted to Python types? Option of defining `sqlalchemy.Table` in code and using it for reflection is unfortunately unsuitable for my case. 

---
Python: 3.11.7, 
SQLite: 3.45.1

**Comment by CaselIT** [comment, marked as answer]
Hi,

No,. there is nothing that converts the default to a python type, since when reflecting a table said default are taken as sql fragments that will be added in the cread table sql.

Note that in general is not trivial to do that, since for dates you could have `NOW` or `now()` there.

**Reply by HardMax71** [reply]
Thanks for quick reply!
